src/libraries/System.IO.FileSystem/tests/ManualTests/ManualTests.cs

```diff
@@ -75,5 +75,65 @@ public static void Throw_FileStreamDispose_WhenRemoteMountRunsOutOfSpace()
                 destinationStream.Dispose();
             });
         }
+
+
+        const long InitialFileSize = 1024;
+
+        [ConditionalFact(nameof(ManualTestsEnabled))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void SetLength_DoesNotAlterPositionWhenNativeCallFails()
+        {
+            /* This test verifies that Position is not altered when SetLength fails when a "disk out of space" error occurs.
+             
+                Setup environment to have a drive with less than 1k available space:
+                - Create an 8mb fixed size VHD (I was having troubles whit smaller sizes).
+                    - Open Computer Management -> Storage -> Disk Management
+                    - Follow these instructions:
+                      https://docs.microsoft.com/en-us/windows-server/storage/disk-management/manage-virtual-hard-disks
+
+                - Restrict the space available in the VHD.
+                    - Create a quota for your user in the VHD created above from cmd:
+                      fsutil quota modify E: 512 512 SYSTEM
+                      fsutil quota modify E: 512 512 YourUser
+
+                - Run the test.
+             */
+
+            using FileStream fs = File.Open("E:/dummy_file.txt", FileMode.OpenOrCreate);
+
+            // Position was less than new Length; should remain the same.
+            fs.Seek(0, SeekOrigin.Begin);
+            VerifySetLength(fs);
+            Assert.True(fs.Position < fs.Length);
+
+            // Position was larger than new Length; should be adjusted to the Length.
+            fs.Seek(InitialFileSize + 1, SeekOrigin.Begin);
+            VerifySetLength(fs);
+            Assert.Equal(fs.Length, fs.Position);
+        }
+
+        private static void VerifySetLength(FileStream fs)
+        {
+            long originalPosition = fs.Position;
+            bool success = false;
+            long size = InitialFileSize;
+
+            while (!success)
+            {
+                try
+                {
+                    Console.WriteLine($"Attempting to write {size} bytes...");
+                    fs.SetLength(size);
+                    Console.WriteLine("Success!");
+                    success = true;
```

Comment:
Why is the loop here necessary?  If the test is configured correctly, it should always succeed, correct?  And if the test is configured incorrectly, wouldn't it be better to fail the test rather than loop infinitely?

Reply:
Oh, sorry, I misread the test.  Nevermind.